java/Oleksii-Riabov/src/com/zoolatech/lecture6/tasks/_3/UserService.java

```diff
@@ -0,0 +1,20 @@
+package com.zoolatech.lecture6.tasks._3;
+
+public class UserService {
+
+    private final UserRepository userRepository;
+
+    public UserService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    public String findUserEmail(String id) {
+        String s = "Email is not found.";
```

Comment:
1. Please find a better naming for the variable - it's not clear what is the meaning of `s`
2. Define the variable as close to it's usage as possible - as for know you'll always create it, even when the record was found

Reply:
"Email is not found." has been moved directly to "return" statement.

Reply:
it might be better just to rethrow an exception or return null in such case; otherwise, `Email is not found.` can be treated as a valid return result of the method

Reply:
Update method of current task to return null instead of String "Email is not found."